src/com/ms/silverking/cloud/storagepolicy/SimpleStoragePolicyCreator.java

```diff
@@ -1,35 +1,28 @@
 package com.ms.silverking.cloud.storagepolicy;
 
 public class SimpleStoragePolicyCreator {
-    public static String    storagePolicyGroupName = "SimplePolicyGroup";
-    public static String    storagePolicyName = "SimpleRackPolicy";
-    
-    public static StoragePolicyGroup createStoragePolicyGroup(int replication) {
-        try {
-            return new PolicyParser().parsePolicyGroup(createDef(replication), 0);
-        } catch (PolicyParseException e) {
-            throw new RuntimeException("Panic");
-        }
-    }
-    
-    private static String createDef(int replication) {
-        String    s;
-        
-        s = "StoragePolicyGroup:"+ storagePolicyGroupName +" {\n"
-           +"    root    Rack:"+ storagePolicyName +"\n"
-           +"}\n"
-           +"\n"
-           +"Rack:"+ storagePolicyName +" {\n"
-           +"    primary {\n"
-           +"        "+ replication +" of Server\n"
-           +"    }\n"
-           +"}\n"
-           +"\n";
-        return s;
-    }
-    
-    public static void main(String[] args) {
-        //System.out.println(createDef(Integer.parseInt(args[0])));
-        System.out.println(createStoragePolicyGroup(Integer.parseInt(args[0])));
+  public static String storagePolicyGroupName = "SimplePolicyGroup";
+  public static String storagePolicyName = "SimpleRackPolicy";
+
+  public static StoragePolicyGroup createStoragePolicyGroup(int replication) {
+    try {
+      return new PolicyParser().parsePolicyGroup(createDef(replication), 0);
+    } catch (PolicyParseException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)